pkg/resourceinterpreter/predefinedconfigurableinterpreter/configurable.go

```diff
@@ -0,0 +1,185 @@
+func (p *PredefinedConfigurableInterpreter) GetDependencies(object *unstructured.Unstructured) (dependencies []configv1alpha1.DependentObjectReference, err error) {
+	luaScript, exist := p.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretDependency)
+	if !exist {
+		return nil, fmt.Errorf("predefined configurable %s interpreter for operation %s not found", configv1alpha1.InterpreterOperationInterpretDependency, object.GroupVersionKind())
+	}
+
+	_ = luaScript
+	// TODO
+	return nil, nil
+}
+
+// ReflectStatus returns the status of the object.
+func (p *PredefinedConfigurableInterpreter) ReflectStatus(object *unstructured.Unstructured) (status *runtime.RawExtension, err error) {
+	luaScript, exist := p.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretStatus)
+	if !exist {
+		return nil, fmt.Errorf("predefined configurable %s interpreter for operation %s not found", configv1alpha1.InterpreterOperationInterpretStatus, object.GroupVersionKind())
+	}
+
+	_ = luaScript
+	// TODO
+	return nil, nil
+}
+
+// InterpretHealth returns the health state of the object.
+func (p *PredefinedConfigurableInterpreter) InterpretHealth(object *unstructured.Unstructured) (healthy bool, err error) {
+	luaScript, exist := p.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretHealth)
+	if !exist {
+		return false, fmt.Errorf("predefined configurable %s interpreter for %q not found", configv1alpha1.InterpreterOperationInterpretHealth, object.GroupVersionKind())
+	}
+
+	_ = luaScript
+	// TODO
+	return false, nil
+}
+
+/* directory structure
+resource_customizations
+├── embed.go
+├── webapp.my.domain                 # Group
+│   └── v1                           # Version
+│       └── Guestbook                # Kind
+│           ├── customizations.lua   # Fixed name
+│           ├── customizations_test.yaml
+│           └── testdata
+└── webapp2.my.domain
+    └── v1
+        └── Guestbook
+            ├── customizations.lua
+            ├── customizations_test.yaml
+            └── testdata
+*/
+func (p *PredefinedConfigurableInterpreter) getPredefinedLuaScripts(objKey string, scriptFile string) (string, error) {
+	data, err := resourcecustomizations.Embedded.ReadFile(filepath.Join(objKey, scriptFile))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", nil
+		}
+		return "", err
+	}
+	return string(data), nil
+}
+func (p *PredefinedConfigurableInterpreter) getConfigMapKey(gvk schema.GroupVersionKind) string {
+	if gvk.Group == "" {
+		return gvk.Kind
+	}
+	return fmt.Sprintf("%s/%s/%s", gvk.Group, gvk.Version, gvk.Kind)
+}
+
+// getInterpreter returns interpreter for specific resource kind and operation.
+func (p *PredefinedConfigurableInterpreter) getInterpreter(kind schema.GroupVersionKind, operation configv1alpha1.InterpreterOperation) (string, bool) {
+	p.lock.RLock()
+	interpreter, ok := p.interpreters[kind][operation]
```

Comment:
When the script file is removed, it's still working?

Reply:
Do you think shall we have a file watch mechanism，updating the configuration?

Reply:
O I get it. These scripts are embedded when compiling. It will not change in running time.
Forget it~